File: src/main/java/com/elissamc/api/ChatSystem/ChatListener.java
package com.elissamc.api.ChatSystem;

import cn.nukkit.Player;
import cn.nukkit.Server;
import cn.nukkit.event.EventHandler;
import cn.nukkit.event.Listener;
import cn.nukkit.event.player.PlayerChatEvent;
import cn.nukkit.plugin.service.RegisteredServiceProvider;
import cn.nukkit.utils.TextFormat;
import me.lucko.luckperms.api.LuckPermsApi;

import java.util.Objects;

public class ChatListener implements Listener {

    private LuckPermsApi api;

    public ChatListener() {
        RegisteredServiceProvider<LuckPermsApi> provider = Server.getInstance().getServiceManager().getProvider(LuckPermsApi.class);
        if (provider != null) {
            api = provider.getProvider();

        }
    }

    private boolean contains(String s) {
        for (ChatFormat cf : ChatFormat.values()) {
            if (cf.name().equals(s))
                return true;
        }
        return false;
    }

    @EventHandler
    public void onChat(PlayerChatEvent event) {
        Player player = event.getPlayer();
        String group = Objects.requireNonNull(api.getUser(player.getUniqueId())).getPrimaryGroup().toUpperCase();
        if (!this.contains(group))
            group = "DEFAULT";
        String s = ChatFormat.valueOf(group).toString();
        s = s.replaceAll("%name", player.getName());
        s = s.replaceAll("%msg", event.getMessage());
        s = TextFormat.colorize(s);
        event.setFormat(s);

    }
}

File: src/main/java/com/elissamc/api/ChatSystem/ChatListener.kt
package com.elissamc.api.ChatSystem

import cn.nukkit.Player
import cn.nukkit.Server
import cn.nukkit.event.EventHandler
import cn.nukkit.event.Listener
import cn.nukkit.event.player.PlayerChatEvent
import cn.nukkit.plugin.service.RegisteredServiceProvider
import cn.nukkit.utils.TextFormat
import me.lucko.luckperms.api.LuckPermsApi
import me.lucko.luckperms.api.User

import java.util.Objects

class ChatListener : Listener {

    private var api: LuckPermsApi? = null

    init {
        val provider = Server.getInstance().serviceManager.getProvider(LuckPermsApi::class.java)
        if (provider != null) {
            api = provider.provider

        }
    }

    private operator fun contains(s: String): Boolean {
        for (cf in ChatFormat.values()) {
            if (cf.name == s)
                return true
        }
        return false
    }

    @EventHandler
    fun onChat(event: PlayerChatEvent) {
        val player = event.player
        var group = Objects.requireNonNull<User>(api!!.getUser(player.uniqueId)).getPrimaryGroup().toUpperCase()
        if (!this.contains(group))
            group = "DEFAULT"
        var s = ChatFormat.valueOf(group).toString()
        s = s.replace("%name".toRegex(), player.name)
        s = s.replace("%msg".toRegex(), event.message)
        s = TextFormat.colorize(s)
        event.format = s

    }
}
